lib/whitehall_importer/embed_body_references.rb

```diff
@@ -34,13 +34,19 @@ def embed_contacts(body, contacts)
     end
 
     def embed_images(body, images)
+      body = replace_one_new_line_before_an_image_with_two(body)
+
       body&.gsub(/!!(\d+)/) do
```

Comment:
Rather than needing an additional similar substitution for this could we do the look up here based on regex last match?

I expect we have a few scenarios too based on whether windows or unix new lines are used. For example:

User input: `\r\n\r\n!!1` Expected Output: `\r\n\r\n[Image: file.jpg]`
User input: `\n\n!!1` Expected Output: `\n\n[Image: file.jpg]`
User input: `\r\n!!1` Expected Output: `\r\n\r\n[Image: file.jpg]`
User input: `\n!!1` Expected Output: `\n\n[Image: file.jpg]`
User input: `!!1` Expected Output: `\n\n[Image: file.jpg]`

And you might be able to manage that with a regex of: `/(\r\n\r\n|\n\n|\r\n|\n)?(!!\d)/` and then you'll have the separator value available in Regexp.last_match[1] to decide whether to change to two new lines in the case of nil or to repeat once in the case of "\n" or "\r\n"

Reply:
@kevindew Thanks for the comments.  I'll look into this.